YOURLS.pot

```diff
@@ -1,30 +1,22 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the YOURLS package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
 #, fuzzy
 msgid ""
 msgstr ""
-"Project-Id-Version: YOURLS 1.9\n"
-"POT-Creation-Date: 2022-04-22 20:19+0200\n"
-"PO-Revision-Date: 2020-04-20 15:07+0100\n"
-"Last-Translator: ozh <ozh@ozh.org>\n"
-"Language-Team: \n"
+"Project-Id-Version: YOURLS 1.9.3-dev\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2025-03-04 12:55+0000\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
 "MIME-Version: 1.0\n"
-"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Type: text/plain; charset=CHARSET\n"
 "Content-Transfer-Encoding: 8bit\n"
-"X-Generator: Poedit 2.3\n"
-"X-Poedit-KeywordsList: yourls__;yourls_e;yourls_s;yourls_se;"
-"yourls_esc_attr__;yourls_esc_html__;yourls_x;yourls_ex;yourls_esc_attr_x;"
-"yourls_esc_html_x;yourls_n:1,2;yourls_nx:1,2;yourls_n_noop:1,2;"
-"yourls_nx_noop:1,2\n"
-"X-Poedit-Basepath: YOURLS\n"
-"Plural-Forms: nplurals=2; plural=n != 1;\n"
-"X-Poedit-SearchPath-0: .\n"
-"X-Poedit-SearchPathExcluded-0: .git\n"
-"X-Poedit-SearchPathExcluded-1: .github\n"
-"X-Poedit-SearchPathExcluded-2: tests\n"
-"X-Poedit-SearchPathExcluded-3: user\n"
-"X-Poedit-SearchPathExcluded-4: css\n"
-"X-Poedit-SearchPathExcluded-5: js\n"
-"X-Poedit-SearchPathExcluded-6: images\n"
-"X-Poedit-SearchPathExcluded-7: includes/vendor\n"
+"Plural-Forms: nplurals=INTEGER; plural=EXPRESSION;\n"
```

Comment:
Most of the changes to this header strike me as.... odd, if not outright unwanted?

Placeholders everywhere for the copyright holder, dates, even `CHARSET` instead of `UTF-8`. 🤔

Reply:
@dgw Yeah, I spent a lot of time understanding these changes, and it actually makes sense.
POT files are template files for PO files (that's the T), no questions on that.
In such, `xgettext` generates headers with placeholder to be filled _by_ the translators.
Indeed, as per templating, it makes more sense to highlight the fields to fill, instead of filling them with the English translations' metadata.

One exception: I'd also prefer the `CHARSET` to be UTF-8. But in the meantime, some language might be outside UTF-8 and this would (possibly) help to parse it. That's very edge, I agree, but as it is the default behavior of `xgettext`, fine for me.

Sounds good? 😊

Reply:
Sure, makes sense to me, thanks. It seems fine as long as it's OK for the `POT-Creation-Date` to get reset. (I don't see a header for `POT-Revision-Date`, only for the `PO`; not sure it matters though.) Removing the Poedit-specific stuff is not a problem if we aren't using that tool any more 😁

How does `Project-Id-Version` work for us when an upstream release happens? It's `1.9.3-dev` right now, naturally, since the next release isn't tagged yet, but the "final" file for 1.10.0 that people start using to create updated translations should have that string, presumably.

Reply:
> How does `Project-Id-Version` work

Autogenerated from there: https://github.com/YOURLS/YOURLS/blob/8cb2ed88258a2de236405f5de4eb8278874dadb3/.github/workflows/update-translations.yml#L35

Yes, I made the action so when a tag is pushed/created, the workflow runs and creates a PR for the said version.

Reply:
Very cool, carry on 👍